pkg/build/pipeline.go

```diff
@@ -227,9 +228,14 @@ func (p *Pipeline) evalRun(ctx *PipelineContext) error {
 	p.With = mutateWith(ctx, p.With)
 	p.dumpWith()
 
+	workdir := "/home/build"
+	if p.WorkDir != "" {
+		workdir = mutateStringFromMap(p.With, p.WorkDir)
+	}
+
 	fragment := mutateStringFromMap(p.With, p.Runs)
 	sys_path := "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
-	script := fmt.Sprintf("#!/bin/sh\nset -e\nexport PATH=%s\n%s\nexit 0\n", sys_path, fragment)
+	script := fmt.Sprintf("#!/bin/sh\nset -e\nexport PATH=%s\nmkdir -p '%s'\ncd '%s'\n%s\nexit 0\n", sys_path, workdir, workdir, fragment)
```

Comment:
(non-blocking) it may help w/ readability in the future for inline scripts like this to use ["raw string literals"](https://go.dev/ref/spec#String_literals) (i.e. using back-ticks and literal newlines instead of `\n`)